hydra-node/src/Hydra/Chain/Direct/Handlers.hs

```diff
@@ -88,22 +90,24 @@ newLocalChainState chainStateAt = do
  where
   getLatest tv = readTVar tv
 
-  pushNew tv cs =
-    modifyTVar tv $ \prev ->
-      cs{previous = Just prev}
+  pushNew tv cs = modifyTVar tv (const cs)
 
-  rollback tv point = do
-    latest <- readTVar tv
-    let rolledBack = go point latest
+  rollback tv point events = do
+    let maybeChainState =
+          events
+            & find
+              ( \ChainStateAt{recordedAt} ->
+                  case recordedAt of
+                    Just recordPoint | recordPoint <= point -> True
+                    _ -> False
+              )
+    -- TODO: using the same as defined in Direct module due to cyclic dependnecy.
+    let initialChainState = ChainStateAt{chainState = Idle, recordedAt = Nothing}
+    let rolledBack = fromMaybe initialChainState maybeChainState
+    -- REVIEW: what should we do with persisted events? prune? replace?
```

Comment:
Would rather keep the previous state in memory and not explore the file content to do the rollback.